File: java/sui-backend/src/main/java/ru/sui/suibackend/service/SuiMetaSettingService.java
package ru.sui.suibackend.service;

import org.springframework.stereotype.Service;
import lombok.RequiredArgsConstructor;
import ru.sui.suientity.repository.suimeta.SuiMetaSettingRepository;

import java.time.*;
import java.util.Date;

@Service
@RequiredArgsConstructor
public class SuiMetaSettingService {
    public Date toDate(LocalDateTime date) {
        return Date.from(date.toInstant(ZoneOffset.UTC));
    }
    private final SuiMetaSettingRepository suiMetaSettingRepository;

    public Integer getInt(String settingKey) {
        return Integer.valueOf(suiMetaSettingRepository.get(settingKey));
    }

    public Long getLong(String settingKey) {
        return Long.parseLong(suiMetaSettingRepository.get(settingKey));
    }
    public Date getDuration(String settingKey) {
        return toDate(LocalDateTime.now().minusDays(getLong(suiMetaSettingRepository.get(settingKey))));
    }
}

File: java/sui-security/base/src/main/java/ru/sui/suisecurity/base/service/SuiMetaSettingService.kt
package ru.sui.suisecurity.base.service

import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service
import ru.sui.suientity.repository.suimeta.SuiMetaSettingRepository
import java.time.LocalDateTime
import java.time.ZoneOffset
import java.util.*

@Service
@RequiredArgsConstructor
class SuiMetaSettingService(
    val suiMetaSettingRepository: SuiMetaSettingRepository
) {
    fun LocalDateTime.toDate(): Date = Date.from(this.toInstant(ZoneOffset.UTC))

    fun getInt(settingKey: String): Int? = suiMetaSettingRepository.get(settingKey).toIntOrNull()

    fun getLong(settingKey: String): Long? = suiMetaSettingRepository.get(settingKey).toLongOrNull()

    fun getDuration(settingKey: String): Date? = getLong(settingKey)?.let { LocalDateTime.now().minusDays(it).toDate() }
}